### PR TITLE
Es/project invite declined

### LINKED ISSUE #233: Project Invitation Declined
## Invitation Declined
- Project Coordinator sends a project invite to a user/device
- As screen shows "Waiting for Device to Accept Invite", invitee declines in the invitation via https://github.com/digidem/mapeo-mobile/issues/1130
- Project Coordinator sees screen of invite declined

Screen Shot 2024-04-06 at 12 31 57 PM

## Todo
- [ ] Display screen as designed
- [ ] Display *Invitation Declined*
- [ ] Display *This device has declined your invitation. They have not joined the project.*
- [ ] Display *Device Name* that declined
- [ ] Create `Close` button

## Design
https://www.figma.com/file/uzynHXR0xgJu19lsCoJyXN/%F0%9F%90%9A-Shell-App?type=design&node-id=1075-5095&mode=design

### PR DESCRIPTION
**Pull request by ErikSin**
Screen to notify user that another user has declined their invite.

closes #233 

image

### CODE REVIEW

File: src/frontend/Navigation/ScreenGroups/AppScreens.tsx [hunk 1]
```diff
@@ -197,9 +196,9 @@ export const createDefaultScreenGroup = (
       options={props => {
         const observationId = props.route.params?.observationId;
         return {
-          headerLeft: props => (
+          headerLeft: headerProp => (
             <CustomHeaderLeftClose
-              headerBackButtonProps={props}
+              headerBackButtonProps={headerProp}
```
Comment (ErikSin) [inline src/frontend/Navigation/ScreenGroups/AppScreens.tsx:201]: This was a ESLint warning (no shadows props).

File: src/frontend/screens/Settings/ProjectSettings/YourTeam/InviteDeclined.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    <View style={styles.container}>
+      <View style={{alignItems: 'center'}}>
+        <ErrorIcon />
+        <Text style={{marginTop: 10, fontSize: 20, fontWeight: 'bold'}}>
+          {formatMessage(m.inviteDeclined)}
+        </Text>
+        <Text style={{marginTop: 10, textAlign: 'center'}}>
+          {formatMessage(m.inviteDeclinedDes)}
+        </Text>
+        <DeviceNameWithIcon {...deviceInfo} style={{marginTop: 10}} />
```
Comment (achou11) [inline src/frontend/screens/Settings/ProjectSettings/YourTeam/InviteDeclined.tsx:55]: based on designs, would add a little more space above this

File: src/frontend/screens/Settings/ProjectSettings/YourTeam/ReviewAndInvite/index.tsx [hunk 3]
```diff
@@ -40,8 +40,13 @@ export const ReviewAndInvite: NativeNavigationComponent<'ReviewAndInvite'> = ({
           navigation.navigate('InviteAccepted', {...route.params});
           return;
         }
+
+        if (val === 'REJECT') {
+          navigation.navigate('InviteDeclined', {...route.params});
```
Comment (achou11) [inline src/frontend/screens/Settings/ProjectSettings/YourTeam/ReviewAndInvite/index.tsx:45]: is object spread here necessary?

```suggestion
          navigation.navigate('InviteDeclined', route.params);
```

**Review by achou11 — APPROVED** [review]

File: src/frontend/screens/Settings/ProjectSettings/YourTeam/ReviewAndInvite/index.tsx (hunk 3 above)
Comment (ErikSin) [reply src/frontend/screens/Settings/ProjectSettings/YourTeam/ReviewAndInvite/index.tsx]: https://github.com/digidem/comapeo-mobile/pull/262/commits/80658c7bcd722a37ecf37660e8acb0d5a72d8269

**Pull request merged**